test/integrations/deployers/DeployerCoordinator.t.sol

```diff
@@ -320,6 +320,85 @@ contract DeployerCoordinatorTest is HyperdriveTest {
         assertEq(deployment.target4, targets[3]);
     }
 
+    function test_deployTarget_frontrun() external {
+        // Deploy a target0 instance as deployer (should succeed).
+        coordinator.deployTarget(DEPLOYMENT_ID, config, new bytes(0), 0, SALT);
+
+        vm.stopPrank();
+        vm.startPrank(bob);
+        // Try to frontrun the target1 deployment with bob
+        vm.expectRevert(
+            IHyperdriveDeployerCoordinator.DeploymentDoesNotExist.selector
+        );
+        coordinator.deployTarget(DEPLOYMENT_ID, config, new bytes(0), 1, SALT);
+        vm.stopPrank();
+    }
+
+    function test_deployTarget_frontrun_success() external {
```

Comment:
This isn't testing anything for target1-4. Additionally, Bob isn't the front-runner. Alice is front-running Bob since she submits her deployment first.

We can fix this by having Alice first deploy all of the targets (as the front-runner), and then leave Bob's loop below. This will demonstrate that Alice can't front-run Bob.